app/migrations/20240809142521-prefill-datasource.cjs

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+const toJson = (text) => ({ user: text });
+
+const getRowsWithUserData = async (queryInterface) => {
+  const rowsWithUserData = await queryInterface.sequelize.query(`select * from "DataSource" where source_type = 'user'`);
+  return rowsWithUserData[0];
+};
+
+function getFormattedRows(rowsWithUserData) {
+  return rowsWithUserData.map(
+    ({
+       transformation_description,
+       dataset_description,
+       methodology_description,
+       dataset_name,
+       ...row
+     }
+    ) => {
+      return {
+        ...row,
+        transformation_description: toJson(transformation_description),
+        dataset_description: toJson(dataset_description),
+        methodology_description: toJson(methodology_description),
+        dataset_name: toJson(dataset_name)
+      };
+    }
+  );
+}
+
+/** @type {import("sequelize-cli").Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    const rowsWithUserData = await getRowsWithUserData(queryInterface);
+    const formattedRows = getFormattedRows(rowsWithUserData);
+    return queryInterface.sequelize.transaction(async (transaction) => {
+      for (const row of formattedRows) {
+        await queryInterface.insert(null, "DataSourceI18n", row, { transaction });
```

Comment:
Not sure if we want to save user data in DataSourceI18n? Let's keep it in DataSource and rename that to UserDataSource in a future refactoring (or just move the 2 relevant fields into ActivityValue tbh, that makes saving easier too). But this should be done in a follow-up PR if possible when we clean up the deprecated fields/ tables in the DB.

Reply:
Created ticket here:
https://openearth.atlassian.net/browse/OC-1431

Reply:
i think the idea was to leave ds alone. if we want to add uds, lets create that and copy